crates/comm/src/peer/mod.rs

```diff
@@ -415,4 +454,11 @@ impl Peer {
         self.peername = Some(peername);
         Ok(())
     }
+
+    async fn get_utc_time() -> DateTime<Utc> {
+        let client = AsyncSntpClient::new();
+        let result = client.synchronize("pool.ntp.org").await.unwrap();
+
+        result.datetime().into_chrono_datetime().unwrap()
+    }
```

Comment:
Probably belongs on `Socket`.

Reply:
(and needs error handling too, otherwise we'll crash as soon as a user loses network connection)

Reply:
how would I call it from inside Peer? should I create functions on Socket that pass the date along?

Reply:
also, why does it make sense for the function to be on `Socket`?

Reply:
got distracted but after implementing more, it seems leaving it in `Peer` is best

Reply:
> how would I call it from inside Peer? should I create functions on Socket that pass the date along?

I wouldn't call it at all from inside `Peer` (see below), I'd spawn a task somewhere in the socket module to periodically fetch the time from NTP and store it in a RwLock, which we can pass an Arc'd copy of to the peer.

> also, why does it make sense for the function to be on `Socket`? I'll move it for now though.

Currently, you're calling `get_utc_time` from `send_available_peers`, meaning this method gets called every time any one of however many peers in the network broadcasts an uptime request, as I understand it, we intend to request in `start_periodic_worker`, see the chat in Discord. 

Considering @YiChenChai's demo situation of 50 AWS nodes, the task runs every 5 seconds for every node in the network, and given a very conservative neighbour count of 3, that's 50*3 = 150 NTP requests every 5 seconds, up to a maximum of 2500 if we are in a fully connected network (unlikely, but not impossible). Sure, these are distributed across the network (done some statistical modelling to see how well this would perform), but given we can just have each node periodically check with NTP what the time is, we can eliminate it with like 20 lines of code.

I think this is less an issue with where the code is currently going and more a sign of an underlying problem with the problem statement. Needs discussion. We can solve the timing issue by moving the NTP call into `Socket`, and I imagine we could add sleep statements to prevent the packet storm we'd otherwise cause.

Reply:
hmm

Reply:
ok i'll finish this up and then work on it

Reply:
It should be okay if we stagger the packets a little, and this is only a group project so it's not that big of a deal, but def something we'd want to fix if we were to take it on. The NTP into Socket should be done though.

Reply:
> (and needs error handling too, otherwise we'll crash as soon as a user loses network connection)

Bumping this again since it's not too hard to fix.

Reply:
I've added error handling I think